### DISCUSSION TITLE
file extension name?

### DISCUSSION THREAD

**Discussion by edwardlee4948** [carbon-language/carbon-lang#1597] [Polls]
What should be the file extension for carbon language?

**Comment by bvoo** [comment]
cbn?

**Reply by linuxias** [reply]
I think, it's good idea! this extension naming is simple and catchy!
If header file extension naming is needed, How about cbh?

**Reply by olamiji-8** [reply]
I think .cb will be better.

**Comment by trotsky1997** [comment]
car，for cars release carbon dioxide，maybe

**Reply by marcauberer** [reply]
And some are made out of carbon fiber 😉

**Reply by trotsky1997** [reply]
> 

and Calcium carbonate can use for serialize

**Comment by ghost** [comment]
I believe .cb is good and simple, I mean I don't need to write another filename as an extension. 😂

**Reply by ItzJustJosh** [reply]
Josh approves this message

**Comment by qascade** [comment]
I saw `.carbon` in a video, is the file extension still not finalised?

**Reply by trotsky1997** [reply]
> I saw `.carbon` in a video, is the file extension still not finalised?

in fact, it can be any name, complier takes them as plain text equaly.

You can change python script's extension to .cc，and python interpreter would also read and work.

**Reply by uladv** [reply]
> > I saw `.carbon` in a video, is the file extension still not finalised?
> 
> in fact, it can be any name, complier takes them as plain text equaly.
> 
> You can change python script's extension to .cc，and python interpreter would also read and work.

Facts. But there should be some standards anyway

**Comment by robiot** [comment]
Id just like to reference this relevant issue here #1574

**Comment by uladv** [comment]
```.cb``` of course

**Comment by ghost** [comment]
.car

why not just call them .car?

did you have a drive license? or are you an very old drivers?
let's drive cars to run.

**Reply by iamtheonewhocodesskyler** [reply]
but you know what? Cars shouldn't run on fossil fuels. Buy a tesla or somthing lol

**Comment by pranampnagle** [comment]
.cbn is ok?

**Comment by im6h** [comment]
`.cb` and no complain.

**Comment by yssource** [comment]
.6c

**Comment by MichaelBonnet** [comment]
.car I think is the move. I've seen .cc elsewhere, but that can be a C++ extension. Also for some reason not a fan of digits in extensions in general.

**Comment by ItzJustJosh** [comment]
Petition to still use the `.carb` extension in the language, not as the main extension, maybe as header files?

**Comment by YagaoDirac** [comment]
Please don't use cb. It's too short, hard for people to search and figure what the hack is the file. 
Extention name is not the same case as in language keywords. Every guy on this planet has a chance to have a .cb file on its computer, and s/he would like to figure what the hack it is.
Make it as long as possible. .carb and .carbon are the correct choices. You type it like no more than 5 time a day, why you care the length of it?

**Reply by AlexDev404** [reply]
Make it long if you want to increase the development cycle. Long file extensions are inappropriate. Also, learning the language alone already requires some research so I do not see any difference in having a short name as they come with many benefits along with shortening the time you spend on creating a file and supporting more filesystems. FAT32 for example has a filename extension limit.

**Comment by chenjpu** [comment]
.c6n

**Comment by alvaromellado** [comment]
1. Use cbn for code file 
2. .carb extension with Header Files

**Reply by MarcoCicognani** [reply]
I hope that Carbon will not need header + impl, but only a single module file, like all the new generation languages

**Reply by Antonioescp** [reply]
I agree with @MarcoCicognani

**Comment by ghost** [comment]
imo .carb look nice

**Reply by AlexDev404** [reply]
Too much carbs 🤣

**Comment by diceptia** [comment]
Cb or cbn nice and simple

**Comment by josh11b** [comment]
#1752 is adding a Q&A explaining why we are not focusing on the file extension at this time.

**Comment by AdiulesonArlen** [comment]
How about .cbl?
Carbon Language.

**Reply by yssource** [reply]
cobol.cbl

**Reply by pop-pic** [reply]
talvez .carb ou .carbn ou .cabo ou .ca ou .carb

**Reply by ghost** [reply]
.cbl is COBOL source code file

**Reply by AlexDev404** [reply]
Taken already by COBOL

**Comment by devrimcavusoglu** [comment]
`.crb` ?

I also like `.cb`, but `.crb` would be better suited I guess a richer version of `.cb`.

**Reply by sethupavan12** [reply]
`.crb` or `.cbn`. But I think the latter is clearer

**Comment by dan36poars** [comment]
my suggestion is that the organic life is made over carbon. the extension name should be *.dna it is much more sense due the language name Carbon. we write programs like the chemistry of the carbon build the life that we know on earth.

**Comment by Rajneeshzytox** [comment]
.cb is easy && simple :)
 Most important: it is short && related to .c || .cpp ;

Or { .cbn }

zytox.cb || zytox.cbn 

Not so important points: 
1, And it is boring to include so many header files, do something about it.
2, can we make a web/ios/Android application with carbon? 
3, is " ; " important?? 
So many

**Reply by Taack** [reply]
Also is `.c6` (even if I do prefer `.6c`), why not choosing `.c6` instead of `.cb` ?

- 6 looks like b;
- `c6` looks cool for everybody having done a little with chemistry;
- It has a real meaning (6 is the atomic number of carbon);
- It has less chances to exist already as file extension;
- Digits are cooler than letters, they are more rare ..

`.c42` has also some meanings (4 being the number of electrons on the second layer, 2 is the pair of electron on the first)

**Comment by cristiancmello** [comment]
Can you use `.krb` (Carbon => "Karbon"). Already exists `.krb` (https://whatext.com/krb) but "The krb file extension is associated with the Pyke, a logic programming solution inspirated by Prolog, created for Python community." ... very old...

Others ideas:

- `.kbn`, `.kbo`, `.kbon`, ....
- `.kbf` (Carbon file)
- `.grf` or `.grp` (Graphene File => Carbon)

Sorry my bad English, however I like this project.

**Comment by ab22593k** [comment]
.crb looks great 💯

**Comment by christianitis** [comment]
After reading some of these comments I am tempted to change my vote from .cb to .carbon

**Comment by rsura-edu** [comment]
`.co2`

**Comment by freedxmgxd** [comment]
The .6c sees great to me, even it's not in the pool.

**Comment by abiriadev** [comment]
I like `cb`, but it is already used for Microsoft clean boot files.

**Comment by tszdanger** [comment]
I guess we can use .bon? No need with a prefix "c".

**Comment by chandlerc** [comment]
Just a reminder, this was made into an issue-for-leads following our evolution process and decided here: https://github.com/carbon-language/carbon-lang/issues/1574#issuecomment-1200068213

**Comment by VELD-Dev** [comment]
I would like to suggest C7H3 (.c7h3) referring to coal

**Comment by qascade** [comment]
I think we should compile all the suggestions and have a poll? But I don't know if it will matter😅. What are the opinions of maintainers on this ?

**Reply by chandlerc** [reply]
The leads already revisited this isue since we opened the project to the public and factored in the new suggestions. As I mentioned above, they made an updated [decision](https://github.com/carbon-language/carbon-lang/issues/1574#issuecomment-1200068213) on #1574. I don't think poll data would be particularly useful at this time.

**Comment by seanmrnda** [comment]
`.cn` or `.cnl` 

`.crbn` ?

What do you guys think?

**Comment by enzv** [comment]
The current _carbon_ extension is even larger than _C++_ extension. Furthermore, modern programming languages uses no more than three characters for its mainly filename extension (e.g, **Kotlin**, **Go**, **Rust**, **Python**). To me, `cb` is the clear winner.

**Reply by AlexDev404** [reply]
I agree. `cb` for source files. But what about headers? I suggest `ch` (Carbon Header) or `mol` (Molecule) if we're being creative. I'd personally prefer `cb` or `ch` though as those are the clear winners IMO.

**Comment by EbadiDev** [comment]
We can do `.cbon` or `.ca`
I like `.cb` and for header `.cbn`

**Reply by shubhujeet** [reply]
Since carbon is the successor of c/c++, I think the .cbon ( pronunciation: c born) extension explains that carbon is the successor of c.
